lib/Mongo/MongoDB.php

```diff
@@ -405,7 +405,12 @@ public function command(array $data, $options = [], &$hash = null)
     {
         try {
             $cursor = new \MongoCommandCursor($this->connection, $this->name, $data);
-            $cursor->setReadPreference($this->getReadPreference());
+
+            if (!empty($data['new']) || !empty($data['upsert'])) {
```

Comment:
My point was that commands should not be detected by the presence of options. As-is, this approach had a few complications:

 * It picks up _any_ [command](https://docs.mongodb.com/manual/reference/command/) that might include a `new` or `upsert` key at the top level. Even if no other command uses these options, it assumes no future non-write command will.
 * The `new` and `upsert` options for `findAndModity` are both boolean options. Using `empty()` here means that the condition is only satisfied if either option is `true`; however, they could both be `false`.
 * Some variations of `findAndModify` do not even use these options (e.g. find and delete).

I don't meant to discourage you with the above criticism. Rather, I want to stress the importance of using a robust solution in library code (be it Doctrine MongoDB, the adapter, or the new extension's PHP library).